src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -2001,6 +2043,22 @@ private static unsafe void ThreadSafeCopy(char* sourcePtr, char[] destination, i
             }
         }
 
+        private static unsafe void ThreadSafeCopy(char* sourcePtr, Span<char> destination, int destinationIndex, int count)
```

Comment:
I'm tentative about those overloads, as they significantly increase code duplication. Still, I have yet to come with an elegant way to factorize them (short of making all codepaths going through the Span version, with a performance penalty). Suggestions are much appreciated